tests/ludwig/modules/test_serialization.py

```diff
@@ -0,0 +1,164 @@
+    data_csv = generate_data(input_features, output_features, os.path.join(tmpdir, "dataset.csv"))
+    model1_config = {
+        "input_features": input_features,
+        "output_features": output_features,
+        "combiner": {"type": "concat", "output_size": 10},
+    }
+    model1 = LudwigModel(model1_config)
+    train_stats1, _, _ = model1.train(dataset=data_csv, output_directory=tmpdir)
+    # Saves pre-trained encoder to file.
+    trained_input_feature = model1.model.input_features[text_input_name]
+    input_feature_encoder = trained_input_feature.encoder_obj
+    saved_path = os.path.join(tmpdir, "text_encoder.h5")
+    serialization.save(input_feature_encoder, saved_path)
+    # Ensures that we can restore encoder from saved path.
+    restored_encoder = serialization.load(saved_path, "cpu")
+    assert restored_encoder is not None
+    # Creates new model referencing the pre-trained encoder.
+    model2_config = {
+        "input_features": [{**input_features[0], "encoder": f"file://{saved_path}", "trainable": True}],
+        "output_features": output_features,
+        "combiner": {"type": "concat", "output_size": 20},
+    }
+    model2 = LudwigModel(model2_config)
+    train_stats2, _, _ = model2.train(dataset=data_csv, output_directory=tmpdir)
+    # Assert that final train loss is lower for model 2 using the pre-trained encoder.
+    # TODO(daniel): Due to randomness, this fails sometimes. Find a better way to test transfer.
+    # Maybe train to a specified performance level and ensure that model2 gets there faster?
+    assert (
+        train_stats2["training"][category_output_name]["loss"][-1]
+        < train_stats1["training"][category_output_name]["loss"][-1]
+    )
+
+
+def test_transfer_learning(tmpdir):
+    torch.random.manual_seed(17)
+    # Trains model 1 on generated dataset.
+    input_features = [text_feature(reduce_output="sum", vocab_size=16, embedding_size=32)]
+    output_features = [category_feature(vocab_size=5)]
+    text_input_name = input_features[0]["name"]  # Auto-generated from random number by text_feature(...)
+    category_output_name = output_features[0]["name"]  # Auto-generated from random number by category_feature(...)
+    data_csv = generate_data(input_features, output_features, os.path.join(tmpdir, "dataset.csv"), 100)
+    model1_config = {
+        "input_features": input_features,
+        "output_features": output_features,
+        "combiner": {"type": "concat", "output_size": 14},
+    }
+    model1 = LudwigModel(model1_config)
+    train_stats1, _, _ = model1.train(dataset=data_csv, output_directory=os.path.join(tmpdir, "model1"))
+    # Saves pre-trained encoder to file.
+    trained_input_feature = model1.model.input_features[text_input_name]
+    input_feature_encoder = trained_input_feature.encoder_obj
+    saved_path = os.path.join(tmpdir, "text_encoder.h5")
+    # Note: I'd like to make the preprocessing metadata a property of the encoder, instead of having to get it from
```

Comment:
I'm in favor of this change. The encoder config should have everything that is needed for the encoder to be properly initialized, which may include computed preprocessing data.

Reply:
We're agreed on that.  Will do metadata refactor in a follow-up PR - I have an open linear issue for this.